addon/index.js

```diff
@@ -32,6 +32,33 @@ function maybeUnwrapProxy(o) {
   return isProxy(o) ? maybeUnwrapProxy(safeGet(o, 'content')) : o;
 }
 
+function tryContent(ctx) {
+  return ctx._content ? ctx._content : ctx.content ? ctx.content : ctx;
+}
+
+function deepNotifyPropertyChange(changeset, path) {
+  let paths = path.split('.');
+  let maybeDynamicPathToNotify = null,
+    lastPath = paths.pop(),
+    current = changeset,
+    i;
+
+  //If the path doesn't exists previously exist inside the CONTENT, this is a dynamic set.
+  let existsInContent = safeGet(changeset[CONTENT], path) !== undefined;
+
+  for (i = 0; i < paths.length; ++i) {
+    const curr = current[paths[i]];
+    if (existsInContent && curr && curr.content === curr.content) {
```

Comment:
`curr` will fail for `falsey` values, correct?  

Do we want to use `getPrototypeOf` and walk the chain checking for `getOwnPropertyDescriptor`?  Or perhaps we just directly check `hasOwnProperty` on the object.

Reply:
Yeah, I think that makes sense